charts/deployment/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if .Values.metrics.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    release: {{ .Values.metrics.prometheus.instance }}
+spec:
```

Comment:
You should add this to spec, because azure monitor have a max length on labels - this will prevent it to be missed:
spec:
  labelLimit: 63
  labelNameLengthLimit: 511
  labelValueLengthLimit: 1023